sqlglot/dialects/snowflake.py

```diff
@@ -258,6 +258,7 @@ class Parser(parser.Parser):
         FUNCTIONS = {
             **parser.Parser.FUNCTIONS,
             "ARRAYAGG": exp.ArrayAgg.from_arg_list,
+            "ARRAY_UNIQUE_AGG": exp.ArrayUniqueAgg.from_arg_list,
```

Comment:
This is not necessary, I believe `ARRAY_UNIQUE_AGG` will be parsed as `ArrayUniqueAgg`

Reply:
Let's add an identity test for Snowflake, i.e. `ARRAY_UNIQUE_AGG(x)` should map to itself

Reply:
I was a little unclear why this mapping exists in the first place. When do you add methods to this mapping?